python/ray/experimental/serve/server.py

```diff
@@ -148,16 +149,24 @@ def __init__(self):
                 await JSONResponse({"error": str(e)})(scope, receive, send)
                 return
 
-        result_object_id_bytes = await (
-            self.serve_global_state.init_or_get_router()
-            .enqueue_request.remote(
-                service=endpoint_name,
-                request_args=(scope, http_body_bytes),
-                request_kwargs=dict(),
-                request_context=TaskContext.Web,
-                request_slo_ms=request_slo_ms))
+        # NOTE: For HTTP requests completely asynchronous call to enqueue
+        #       request is not supported!
+        request_params = RequestParams(
+            endpoint_name, TaskContext.Web, request_slo_ms=request_slo_ms)
 
-        result = await ray.ObjectID(result_object_id_bytes)
+        # TODO(alind): File a Ray issue if args contain b"" it is not
+        #              received.
+        # Hence enclosing http_body_bytes inside a list.
```

Comment:
@simon-mo the reason for `[0]` is because of Ray Issue

Reply:
Yeah thanks!